credentials/credentials.go

```diff
@@ -126,6 +126,10 @@ func (t TLSInfo) AuthType() string {
 	return "tls"
 }
 
+func (t TLSInfo) ConnState() tls.ConnectionState {
```

Comment:
I think a getter makes more sense here; you don't want to allow mutation of this state.